index.js

```diff
@@ -7,26 +7,36 @@ const mergeTrees = require('broccoli-merge-trees');
 let htmlCache = null;
 
 module.exports = {
-
   name: 'ember-cli-favicon',
 
-  included(app) {
-    let options = app.options;
+  included(parent) {
+    // Support for fingerprint
+    parent.options.fingerprint = parent.options.fingerprint || {};
+    parent.options.fingerprint.exclude = parent.options.fingerprint.exclude || [];
+    parent.options.fingerprint.exclude.push('apple-touch-icon', 'favicon', 'mstile');
+
+    // Set success callback
+    parent.options.favicons = parent.options.favicons || {};
 
-    let fingerprint = options.fingerprint = options.fingerprint || {};
-    fingerprint.exclude = fingerprint.exclude || [];
-    fingerprint.exclude.push('apple-touch-icon', 'favicon', 'mstile');
+    let currentCallback = parent.options.favicons.htmlCallback || function() {};
 
-    this.options = options.favicons || {};
-    this.options.htmlCallback = function(html) {
+    parent.options.favicons.htmlCallback = function(html) {
       htmlCache = html;
+      return currentCallback(...arguments);
     };
+
+    this.parentOptions = parent.options;
+
+    return this._super.included(parent);
+  },
+
+  treeForPublic(tree) {
+    let faviconTree = new Favicons(this.parentOptions.trees.public, this.parentOptions.favicons);
+    return mergeTrees([ faviconTree, tree ].filter((v) => v), { overwrite: true });
```

Comment:
One nice little take on `.filter((v) => v)` is `.filter(Boolean)` instead. It's a little more explicit and concise. But if you prefer this syntax, don't bother changing. Just throwing that out there.

Reply:
Wow cool thing I didn't know that's possible. :)